### PR TITLE
Rename aggrsssive fusion to fuse-multi-use

### PR DESCRIPTION
**Pull request by pzread**
Since most of the aggressive fusion has been enabled by default. This option now only controls the fusion for multi-use ops. Rename it and keep it off by default.

benchmarks: all

### CODE REVIEW

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp [hunk 1]
```diff
@@ -464,7 +461,7 @@ static bool isFusableWithConsumer(
 
   // Check if the iteration spaces of the producer and consumer are same.
   // TODO: This is unnecessary requirement, but needed to pass tests right now
-  if (!aggressiveFusion) {
+  if (!fuseMultiUse) {
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp]: Just as an experiment... can you try removing lines `464`-`470` and see what happens. This is not really related to `fuseMultiUse`.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp [hunk 2]
```diff
@@ -541,24 +538,23 @@ static bool isFusableWithProducer(
   auto consumerLinalgOp = cast<linalg::LinalgOp>(consumer);
   if (consumerLinalgOp.isDpsInput(&operand)) {
     // Only fuse on inputs if both ops are generic ops.
-    if (!aggressiveFusion || !isa<linalg::GenericOp>(consumer) ||
+    if (!fuseMultiUse || !isa<linalg::GenericOp>(consumer) ||
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp]: This one too is not related to `fuseMultiUse`... Can we try making this 
```
if (!isa<linalg::GenericOp>(consumer) || !isa<linalg::GenericOp>(producer)) {
  return false;
}
```

and see if there are any issues....

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Thanks! Looks mostly OK. There are a couple of things I see here that are not really related to `multiUse`. Maybe those can be turned on by default separately before we make this change. I left a comment in the relevant places.

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp (hunk 1 above)
Comment (pzread) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp]: Landed in #12589

File: compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp (hunk 2 above)
Comment (pzread) [reply compiler/src/iree/compiler/Dialect/Flow/Transforms/FormDispatchRegions.cpp]: Landed in #12589

**Comment by MaheshRavishankar** [comment]
Looks good to me. So whats the final number of dispatches we end up with here?

**Review by MaheshRavishankar — APPROVED** [review]

**Comment by pzread** [comment]
> Looks good to me. So whats the final number of dispatches we end up with here?

The previous https://github.com/openxla/iree/pull/12589 didn't change the numbers of dispatches. This one only renames the options, so I also don't expect the changes. I can double check.

**Comment by MaheshRavishankar** [comment]
> > Looks good to me. So whats the final number of dispatches we end up with here?
> 
> The previous #12589 didn't change the numbers of dispatches. This one only renames the options, so I also don't expect the changes. I can double check.

Just want to document what it is with this change (after culmination of all changes) to mark where we leave this off for now.

**Comment by pzread** [comment]
> > > Looks good to me. So whats the final number of dispatches we end up with here?
> > 
> > 
> > The previous #12589 didn't change the numbers of dispatches. This one only renames the options, so I also don't expect the changes. I can double check.
> 
> Just want to document what it is with this change (after culmination of all changes) to mark where we leave this off for now.

All architectures gave out the same number of dispatches (no change before/after)

|Model                |Dispatches (After)|
|---------------------|------------------|
|BertForMaskedLMTF    |213               |
|BertLargeTF          |413               |
|DeepLabV3_fp32       |78                |
|EfficientNet_int8    |137               |
|EfficientNetV2STF    |278               |
|MiniLML12H384Uncased |211               |
|MobileBertSquad_fp16 |752               |
|MobileBertSquad_fp32 |752               |
|MobileBertSquad_int8 |1452              |
|MobileNetV1_fp32     |57                |
|MobileNetV2_fp32     |74                |
|MobileNetV3Small_fp32|98                |
|MobileSSD_fp32       |121               |
|PersonDetect_int8    |87                |
|PoseNet_fp32         |48                |
|Resnet50TF           |75                |

**Comment by MaheshRavishankar** [comment]
Thanks! If you get time can you post the number with the multiuse flag turned on.

**Comment by pzread** [comment]
> Thanks! If you get time can you post the number with the multiuse flag turned on.

For x86_64 target with default flags

|Model                |Before|After (with --iree-fuse-multi-use)|
|---------------------|------|-----|
|BertForMaskedLMTF    |213   |187  |
|BertLargeTF          |413   |364  |
|ClipTextSeqLen64PT   |197   |N/A (don't compile)  |
|DeepLabV3_fp32       |78    |78   |
|EfficientNet_int8    |137   |137  |
|EfficientNetV2STF    |278   |275  |
|MiniLML12H384Uncased |211   |184  |
|MobileBertSquad_fp16 |752   |751  |
|MobileBertSquad_fp32 |752   |727  |
|MobileBertSquad_int8 |1452  |1427 |
|MobileNetV1_fp32     |57    |57   |
|MobileNetV2_fp32     |74    |74   |
|MobileNetV2_int8     |196   |196  |
|MobileNetV3Small_fp32|98    |96   |
|MobileSSD_fp32       |121   |121  |
|PersonDetect_int8    |87    |87   |
|PoseNet_fp32         |48    |48   |
|Resnet50TF           |75    |75   |
|Unet2dPT             |985   |935  |
